sniffer.py

```diff
@@ -228,20 +218,22 @@ def main():
                     timestamp_sec = timebase_sec + timestamp_usec / 1000000
                     timestamp_usec = timestamp_usec % 1000000
 
-                    if options.rssi:
-                        pkt = pkt[:-2] + chr(metadata[0] & 0xff) + chr(0x80)
+                    rssi = chr(metadata[0] & 0xff) + chr(0x80)
 
                 # Some old version NCP doesn't contain timestamp information in metadata
                 else:
                     timestamp = datetime.utcnow() - epoch
                     timestamp_sec = timestamp.days * 24 * 60 * 60 + timestamp.seconds
                     timestamp_usec = timestamp.microseconds
 
-                    if options.rssi:
-                        pkt = pkt[:-2] + chr(127) + chr(0x80)
-                        sys.stderr.write("WARNING: failed to display RSSI, please update the NCP version\n")
+                    rssi = chr(127) + chr(0x80)
+                    sys.stderr.write("WARNING: failed to display RSSI, please update the NCP version\n")
+
+                if options.tap:
+                    pkt = pcap.encode_frame(pkt, timestamp_sec, timestamp_usec, options.rssi, options.crc, rssi, metadata)
+                else:
+                    pkt = pcap.encode_frame(pkt, timestamp_sec, timestamp_usec, options.rssi, options.crc, rssi)
```

Comment:
how about directly passing options and metadata into `encode_frame()` and in the body of `encode_frame()`, apply different framing method according to the `_dlt` in the pcap header?